dart/lib/src/sentry_options.dart

```diff
@@ -155,11 +159,13 @@ class SentryOptions {
 
   set debug(bool newValue) {
     _debug = newValue;
-    if (_debug == true && logger == noOpLogger) {
-      _logger = _debugLogger;
+    if (_debug == true &&
+        (logger == noOpLogger || diagnosticLogger?.logger == noOpLogger)) {
```

Comment:
Why the additional `diagnosticLogger?.logger == noOpLogger)` here? Removed and running `sentry_options_test.dart` all tests pass, so we are not testing this condition or it is always the same to logger.

Reply:
if you change the debug flag multiple times the logger will only be wrapped by the diagnosticLogger and will neither be noOpLogger or debugLogger directly. so this is basically a way to handle the edge case where the user sets the debug to true and false multiple times. it's an edge case though